src/services/data_model_engine.js

```diff
@@ -208,4 +208,19 @@ export default class DataModelEngine {
     await this.entityRepository.storeBundle(bundle);
     return bundle;
   }
+
+  async updateShelteringExpirationDate(bundleId) {
+    const expirationDate = await this.uploadRepository.expirationDate(bundleId);
+    await this.entityRepository.storeBundleShelteringExpirationDate(bundleId, expirationDate);
+  }
+
+  async cleanupBundles() {
+    const allBundleIds = await this.entityRepository.getExpiredBundleIds();
+    const isSheltering = await Promise.all(allBundleIds.map((bundleId) => this.uploadRepository.isSheltering(bundleId)));
+    const toBeRemoved = allBundleIds.filter((bundleId, ind) => !isSheltering[ind]);
+    const toBeUpdated = allBundleIds.filter((bundleId, ind) => isSheltering[ind]);
+    await this.entityRepository.deleteBundles(toBeRemoved);
+    await Promise.all(toBeUpdated.map((bundleId) => this.updateShelteringExpirationDate(bundleId)));
```

Comment:
Can't imagine situation when sheltering particular bundle, by particular shelterer changes its expiration date 🤔 🤔 🤔 
Once Atlas won a challenge, he shelters it for certain amount of time and that's it.
Initial putting expiration date into metadata should rather be performed immediately after successful challenge resolution.

Reply:
It can be not set by some reason

Reply:
`allBundleIds` is somewhat misleading.